File: app/src/main/java/fr/clerc/tp3/EmployeeDetailsActivity.java
package fr.clerc.tp3;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import fr.clerc.tp3.model.Employee;
import fr.clerc.tp3.model.EmployeeData;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

import static fr.clerc.tp3.MainActivity.EMPLOYEE_ID;

public class EmployeeDetailsActivity extends AppCompatActivity {

    private TextView name;
    private TextView age;
    private TextView salary;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_employee_detail);

        name = findViewById(R.id.employee_name);
        age = findViewById(R.id.employee_age);
        salary = findViewById(R.id.employee_salary);

        Toolbar toolbar = findViewById(R.id.employeeDetailToolbar);
        toolbar.setTitle(getString(R.string.employee_label));
        setSupportActionBar(toolbar);

        toolbar.setNavigationIcon(R.drawable.ic_back);
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });

        Bundle bundle = getIntent().getExtras();
        if(bundle != null) {
            int id = bundle.getInt(EMPLOYEE_ID, -1);
            if (id >= 0) fetchAgencyData(id);
        }
    }

    private void fetchAgencyData(int id) {
        EmployeesService employeesService = RetrofitClient.getInstance().create(EmployeesService.class);
        employeesService.getEmployee(id).enqueue(new Callback<Employee>() {

            @Override
            public void onResponse(Call<Employee> call, Response<Employee> response) {
                if(response.isSuccessful() && response.body() != null) {
                    //Manage data
                    displayData(response.body().getData());
                } else {
                    Toast.makeText(getApplicationContext(), getString(R.string.app_error), Toast.LENGTH_LONG).show();
                }
            }

            @Override
            public void onFailure(Call<Employee> call, Throwable t) {
                //Manage errors
            }
        });
    }

    private void displayData(EmployeeData employee) {
        name.setText(employee.getEmployeeName());
        age.setText(String.valueOf(employee.getEmployeeAge()));
        salary.setText(String.valueOf(employee.getEmployeeSalary()));
    }
}

File: app/src/main/java/fr/clerc/tp3/kotlin/EmployeeDetailsActivity.kt
package fr.clerc.tp3.kotlin

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import fr.clerc.tp3.R
import fr.clerc.tp3.RetrofitClient
import fr.clerc.tp3.kotlin.model.Employee
import fr.clerc.tp3.kotlin.model.EmployeeData
import kotlinx.android.synthetic.main.activity_employee_detail.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class EmployeeDetailsActivity: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_employee_detail)

        employeeDetailToolbar.title = getString(R.string.employee_label)
        setSupportActionBar(employeeDetailToolbar)

        employeeDetailToolbar.setNavigationIcon(R.drawable.ic_back)
        employeeDetailToolbar.setNavigationOnClickListener {
            finish()
        }

        val id = intent.getIntExtra(MainActivity.EMPLOYEE_ID, 0)
        if (id >= 0) fetchEmployeeData(id)
    }

    private fun fetchEmployeeData(id: Int) {
        val employeesService = RetrofitClient.getInstance().create(EmployeesService::class.java)
        employeesService.getEmployee(id).enqueue(object : Callback<Employee?> {
            override fun onResponse(call: Call<Employee?>, response: Response<Employee?>) {
                if (response.isSuccessful && response.body() != null) { //Manage data
                    val employee = response.body()
                    if(employee != null) {
                        displayData(employee.data)
                    } else {
                        Toast.makeText(applicationContext, getString(R.string.app_error), Toast.LENGTH_LONG).show()
                    }
                }
            }

            override fun onFailure(call: Call<Employee?>, t: Throwable) { //Manage errors
            }
        })
    }

    private fun displayData(data: EmployeeData) {
        employee_name.text = data.employeeName
        employee_age.text = data.employeeAge.toString()
        employee_salary.text = data.employeeSalary.toString()
    }
}